docs/agent-reference/agent-core/configuration-manager.md

```diff
@@ -0,0 +1,95 @@
+# Device Update Agent Configuration
+
+## APIs
+
+```c
+// A config object handle.
+typedef void* CONFIG_HANDLE
+
+// Instancitate a config object from specified file.
+CONFIG_HANDLE LoadConfig(const char * configFilePath);
+
+// Free all resources allocated for the config object.
+void FreeConfig(CONFIG_HANDLE config);
+
+// Returns value of the specified config name.
+// (Output is a json string).
+const char* GetValue(const char* name);
+
+```
+
+## Configuration File Format
+
+See example `du-config.json` file format below:
+
+```json
+
+"enabledIoTHubTracing" : true,
+"logLevel" : 0,
+
+"aduShellTrustedUsers": ["adu", "do"],
+
+"agents":[
+  {
+      "name":"host-update",
+      "runas":"adu",
+      "connectionInfoSource":"AIS",
+      "connectionData":{
+          "connectionType" : "module",
+          "moduleName" : "ioTHubDeviceUpdate"
+      },
+
+      "aduc_manufacturer" : "Contoso",
+      "aduc_model" : "Smart-Box"
+  },
+
+  {
+      "name":"leaf-update",
+      "runas":"adu",
+      "connectionInfoSource":"string",
+      "connectionData":{
+          "connectionString" : "HostName=..."
+      },
+
+      "aduc_manufacturer" : "Fabrikam",
+      "aduc_model" : "Camera"
+  }
+],
+
+"contentHandlers":[
```

Comment:
Oh, swupdate, apt and mcu are all three shared libraries. (claps) 
What are the SHA for?  Do the users set the SHA in the config file themselves?

Reply:
Current thought, either in config file, or add registration file(s) under /etc/adu/content-handlers.d/ directory.